src/Controllers/cli_controller.py

```diff
@@ -1,22 +1,26 @@
-
 from Models.uml_diagram import UML_Diagram
-
 from prompt_toolkit import prompt
 from Views.cli_view import CLI_View
+from prompt_toolkit.completion import NestedCompleter
+
 
 class CLI_Controller:
 
     def __init__(self):
         self._view = CLI_View()
-    
+
     def request_update(self):
-         return prompt("Command: ", completer=self._view._completer).strip()
+        self.setup_autocomplete()
+        return prompt("Command: ", completer=self._completer).strip()
 
     def draw(self, diagram: UML_Diagram):
         pass
 
-#=========================CLI Specific Parseing=========================#  
-    def parse_list_cmd(self, d:UML_Diagram, tokens:list[str]):
+    def get_diagram(self, diagram: UML_Diagram):
+        self._diagram = diagram
```

Comment:
Duplicating the diagram is a huge risk for security. It also breaks the existing program design. One solution could be making the diagram singleton, which would allow you to just construct it whenever you need it and get a reference to the existing one.

Reply:
If you are giving a class a field (self. indicates that it is a field) it needs to be listed in the __init__ method and constructed when __init__ is run. This is python convention and also just good code style. Tell people what they are creating when they construct an object.

Reply:
We are grabbing a reference of the diagram, which I discussed with you before. I can't put it in the init because it is a call made when uml_controller runs it's init